pkg/ddl/backfilling.go

```diff
@@ -752,8 +752,8 @@ type localRowCntListener struct {
 }
 
 func (s *localRowCntListener) Written(rowCnt int) {
-	s.curPhysicalRowCnt += int64(rowCnt)
-	s.reorgCtx.setRowCount(s.prevPhysicalRowCnt + s.curPhysicalRowCnt)
+	newCurPhysicalRowCnt := atomic.AddInt64(&s.curPhysicalRowCnt, int64(rowCnt))
```

Comment:
The root cause is concurrent calling of `Written`, although we make `s.curPhysicalRowCnt` concurrent-safe, below `setRowCount` may set to the smaller `newCurPhysicalRowCnt` value of multiple concurrent calls.

Maybe add a lock? /cc @tangenta

Reply:
agree. actually i am a little bit confuse the usage of `curPhysicalRowCnt`, since there is no initialization and other reference, just equal to `rowCnt`. 

Maybe it can simplify to `s.reorgCtx.setRowCount(s.prevPhysicalRowCnt + int64(rowCnt))`?